zerver/worker/queue_processors.py

```diff
@@ -819,7 +821,20 @@ def send_email(self, event: Dict[str, Any]) -> None:
             del copied_event["failed_tries"]
         handle_send_email_format_changes(copied_event)
         self.connection = initialize_connection(self.connection)
-        send_email(**copied_event, connection=self.connection)
+        if settings.SMTP_MAX_CONNECTION_MINUTES == 0 or self.connection_open_timestamp == 0:
+            self.connection.open()
+            self.connection_open_timestamp = int(time.time())
+            send_email(**copied_event, connection=self.connection)
+            self.connection.close()
+        else:
+            if (
+                int(time.time()) - self.connection_open_timestamp
+                > settings.SMTP_MAX_CONNECTION_MINUTES * 60
+            ):
+                self.connection.close()
+                self.connection.open()
+                self.connection_open_timestamp = int(time.time())
+                send_email(**copied_event, connection=self.connection)
```

Comment:
This should probably be done by adjusting/extending the `initialize_connection` interface.

Reply:
Made the requested changes